File: 20220720/모법답안/20220720_04.js/UniqueOccurrencesNumber_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.easy.UniqueOccurrencesNumber.UniqueOccurrencesNumber import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        arr = [1, 3, 5, 4, 3, 1, 1]

        # when
        result = solution(arr)

        # then
        self.assertEqual(False, result)

    def test_2(self):
        # given
        arr = [6, 18, 12, 3, 1, 12]

        # when
        result = solution(arr)

        # then
        self.assertEqual(False, result)

    def test_3(self):
        # given
        arr = [5, 18, 2, 1, 7, 13]

        # when
        result = solution(arr)

        # then
        self.assertEqual(False, result)

    def test_4(self):
        # given
        arr = [5, 1, 1, 6, 4, 6, 4, 4, 1, 1]

        # when
        result = solution(arr)

        # then
        self.assertEqual(True, result)

    def test_5(self):
        # given
        arr = [1]

        # when
        result = solution(arr)

        # then
        self.assertEqual(True, result)

    def test_6(self):
        # given
        arr = [10, 9, 9, 8, 8, 8, 7, 7, 7, 7, 6, 6, 6, 6, 6, 6]

        # when
        result = solution(arr)

        # then
        self.assertEqual(True, result)


if __name__ == '__main__':
    unittest.main()

File: 20220720/모법답안/20220720_04.js/UniqueOccurrencesNumber_test.js
import assert from 'assert'
import solution from './UniqueOccurrencesNumber.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const arr = [1, 3, 5, 4, 3, 1, 1]

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(false, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const arr = [6, 18, 12, 3, 1, 12]

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(false, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const fruits = [5, 18, 2, 1, 7, 13]

    // when
    const result = solution(fruits)

    // then
    assert.deepStrictEqual(false, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const fruits = [5, 1, 1, 6, 4, 6, 4, 4, 1, 1]

    // when
    const result = solution(fruits)

    // then
    assert.deepStrictEqual(true, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const fruits = [1]

    // when
    const result = solution(fruits)

    // then
    assert.deepStrictEqual(true, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const fruits = [10, 9, 9, 8, 8, 8, 7, 7, 7, 7, 6, 6, 6, 6, 6, 6]

    // when
    const result = solution(fruits)

    // then
    assert.deepStrictEqual(true, result)
    done();
  });
})
